dali/c_api/c_api.cc

```diff
@@ -105,20 +105,32 @@ void daliCreatePipeline(daliPipelineHandle* pipe_handle,
     pipe->SetQueueSizes(cpu_prefetch_queue_depth, gpu_prefetch_queue_depth);
   }
   pipe->Build();
-  pipe_handle->pipe = reinterpret_cast<void*>(pipe);
+  pipe_handle->pipe = reinterpret_cast<void *>(pipe);
   pipe_handle->ws = new dali::DeviceWorkspace();
   CUDA_CALL(cudaStreamCreateWithFlags(&pipe_handle->copy_stream, cudaStreamNonBlocking));
 }
 
-void daliPrefetchUniform(daliPipelineHandle* pipe_handle, int queue_depth) {
-  dali::Pipeline* pipeline = reinterpret_cast<dali::Pipeline*>(pipe_handle->pipe);
+
+void daliDeserializeDefault(daliPipelineHandle *pipe_handle, const char *serialized_pipeline,
+                            int length) {
+  dali::Pipeline *pipe = new dali::Pipeline(std::string(serialized_pipeline, length));
+  pipe->Build();
```

Comment:
If this throws, resources will leak.
The best option is to use a unique_ptr and release it in the next line.

Reply:
Done